src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -98,6 +98,23 @@ public virtual async Task<string> SaveObservationAsync(ILookupTemplate<IFhirTemp
             return result.Id;
         }
 
+        public override Task<string> CheckHealth()
+        {
+            SearchParams q = new SearchParams()
+                .ForSubject<Model.Patient>("IoMTHealthCheck")
+                .SetCount(1);
+
+            var result = _client.Search(q);
+            if (result != null && result.GetType() == typeof(Model.Bundle))
+            {
+                return Task.FromResult("Healthy");

```

Comment:
We should record the result as a metric and include the timestamp as a dimension.  It would be even better to record the http status code as well so we can differentiate between unauthorized access and the service is not available.

Reply:
Yeah, we should. I was thinking of doing the logging as part of a try/catch in Functions.cs in a non OSS repo though. I didn't do the logging here since the metrics logging probably isn't needed for OSS.

The http status code is revealed in the exception when _client.search fails, so if we wrap CheckMetrics in a try/catch in Functions.cs then we should have access to the status code (It won't be discrete but it will be in the exception text).

Reply:
I agree with Dustin. If the result is binary, we are putting the DRI in a bit of a pickle - All they see is that something is broken.

Reply:
Regarding the merits of recording the metrics in OSS, if it is done around the ILogger interface we should be in a good position to inject additional logging channels when appropriate.

I am also fine with the recording being further up the stack but the result from the function needs to return the detail necessary still.